File: algorithms/sort/insertionSort.test.ts
import insertionSort from './insertionSort.ts'

function testInsertionSort() {
    let nums = [5, -2, 12, 0, 9], expected = [-2, 0, 5, 9, 12]
    insertionSort(nums)
    console.assert(JSON.stringify(nums) === JSON.stringify(expected), `expect insertionSort(...) to be: ${expected}, got instead: ${nums}`)

    nums = [1, -2, 10, -6], expected = [-6, -2, 1, 10]
    insertionSort(nums)
    console.assert(JSON.stringify(nums) === JSON.stringify(expected), `expect insertionSort(...) to be: ${expected}, got instead: ${nums}`)
}

testInsertionSort()

File: algorithms/sort/insertion_sort_test.go
package sort

import (
	"reflect"
	"testing"
)

func TestInsertionSortInt(t *testing.T) {
	expected, nums := []int{-2, 0, 5, 9, 12}, []int{12, 0, 9, -2, 5}
	InsertionSortInt(nums, func(prev, next int) bool {
		return prev < next
	})
	if !reflect.DeepEqual(expected, nums) {
		t.Errorf("expect InsertionSort(...) to be [%v], got instead [%v]", expected, nums)
	}

	expected, nums = []int{10, 1, -2, -6}, []int{1, -2, 10, -6}
	InsertionSortInt(nums, func(prev, next int) bool {
		return prev > next
	})
	if !reflect.DeepEqual(expected, nums) {
		t.Errorf("expect InsertionSort(...) to be [%v], got instead [%v]", expected, nums)
	}
}
